lib/puppet/functions/nebula/user_group_membership.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+# Copyright (c) 2018 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+
+Puppet::Functions.create_function(:'nebula::user_group_membership') do
```

Comment:
It might be helpful to have a more explicit comment/example that it returns a hash where the keys are users and the values are the list of all groups for the user.

It might be nice to write this in a more functional style, but if this works I'm inclined to leave it alone (especially given the vagaries of the interaction of puppet data types & ruby)

Reply:
It also would be nice to have tests specifically for this function, if for no other reason than it would aid later refactoring.

Reply:
At first I didn't want to do this because I was thinking of it as a private method, but I realized it could pretty easily be generalized into `inverted_hashlist()`, so I went ahead and did that. Also, that way, the particular hiera lookup is named in a `.pp` file instead of a `.rb` file, which feels better to me. Also also, Bryan taught me about the `tap` method, so it looks a lot more functional now.